### ISSUE TITLE
pathlib.glob('**') returns only directories

### ISSUE THREAD

**Issue by a0f151b7-f5e3-4789-9cc8-8192d970bc91** [python/cpython#70303]
BPO | [26115](https://bugs.python.org/issue26115)
--- | :---
Nosy | @pitrou

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2016-01-14.22:26:27.607>
created_at = <Date 2016-01-14.21:20:32.100>
labels = ['type-bug', 'invalid']
title = "pathlib.glob('**') returns only directories"
updated_at = <Date 2016-02-04.07:25:04.510>
user = 'https://bugs.python.org/jitterman'
```

bugs.python.org fields:
```python
activity = <Date 2016-02-04.07:25:04.510>
actor = 'SilentGhost'
assignee = 'none'
closed = True
closed_date = <Date 2016-01-14.22:26:27.607>
closer = 'SilentGhost'
components = []
creation = <Date 2016-01-14.21:20:32.100>
creator = 'jitterman'
dependencies = []
files = []
hgrepos = []
issue_num = 26115
keywords = []
message_count = 3.0
messages = ['258223', '258237', '259537']
nosy_count = 3.0
nosy_names = ['pitrou', 'SilentGhost', 'jitterman']
pr_nums = []
priority = 'normal'
resolution = 'not a bug'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue26115'
versions = ['Python 3.5']
```

</p></details>

### Linked PRs
* gh-105413
* gh-114684

**Comment by barneygale** [comment]
Re-opening - this is a valid issue.

**Comment by terryjreedy** [comment]
The [Path.glob](https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob) says that patterns are the same as for [fnmatch](https://docs.python.org/3/library/fnmatch.html#module-fnmatch), except that '**' is always recursive.  The fnmatch doc says nothing about '**'.  The [glob dob](https://docs.python.org/3/library/glob.html#module-glob) says glob is essentially scandir+fnmatch, and that the behavior of '**' depends on options selected.
> If recursive is true, the pattern “**” will match any files and zero or more directories, subdirectories and symbolic links to directories. If the pattern is followed by an [os.sep](https://docs.python.org/3/library/os.html#os.sep) or [os.altsep](https://docs.python.org/3/library/os.html#os.altsep) then files will not match.

>If include_hidden is true, “**” pattern will match hidden directories.)

To me, the current Path.glob '**' is a behavior that should be fixed.

**Comment by barneygale** [comment]
Fixed in 3.13 / [fda7445](https://github.com/python/cpython/commit/fda7445ca50b892955fc31bd72a3615fef1d70c6) / #114684
